src/Analyzers/CSharp/CodeFixes/MakeMemberRequired/CSharpMakeMemberRequiredCodeFixProvider.cs

```diff
@@ -0,0 +1,133 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.MakeMemberRequired;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.MakeMemberRequired), Shared]
+[ExtensionOrder(Before = PredefinedCodeFixProviderNames.DeclareAsNullable)]
+internal sealed class CSharpMakeMemberRequiredCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+{
+    private const string CS8618 = nameof(CS8618); // Non-nullable variable must contain a non-null value when exiting constructor. Consider declaring it as nullable.
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public CSharpMakeMemberRequiredCodeFixProvider()
+    {
+    }
+
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+        ImmutableArray.Create(CS8618);
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var document = context.Document;
+        var span = context.Span;
+        var cancellationToken = context.CancellationToken;
+
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        // Required members are available in C# 11 or higher
+        if (root.GetLanguageVersion() < LanguageVersion.CSharp11)
+            return;
+
+        var node = root.FindNode(span);
+
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+        var fieldOrPropertySymbol = semanticModel.GetDeclaredSymbol(node, cancellationToken);
+        if (fieldOrPropertySymbol is IPropertySymbol propertySymbol)
+        {
+            var setMethod = propertySymbol.SetMethod;
+
+            // Property must have a `set` or `init` accessor in order to be able to be required
+            if (setMethod is null)
+                return;
+
+            var containingTypeVisibility = propertySymbol.ContainingType.GetResultantVisibility();
+            var minimalAccessibility = (Accessibility)Math.Min((int)propertySymbol.DeclaredAccessibility, (int)setMethod.DeclaredAccessibility);
+
+            if (!CanBeAccessed(containingTypeVisibility, minimalAccessibility))
+                return;
+
+            RegisterCodeFix(context, CSharpCodeFixesResources.Make_property_required, nameof(CSharpCodeFixesResources.Make_property_required));
```

Comment:
you register the codefix before being sure that the fixer code can run without a problem.  specifically that node you get back is the right shape.